src/sentry/api/endpoints/project_users.py

```diff
@@ -37,21 +49,95 @@ def get(self, request: Request, project) -> Response:
             project_id=project.id,
             endpoint="sentry.api.endpoints.project_users.get",
         )
-        queryset = EventUser.objects.filter(project_id=project.id)
+        now = timezone.now()
+        where_conditions = [
+            Condition(Column("project_id"), Op.EQ, project.id),
+            Condition(Column("timestamp"), Op.GTE, project.date_added),
+            Condition(Column("timestamp"), Op.LT, now),
+        ]
         if request.GET.get("query"):
             try:
                 field, identifier = request.GET["query"].strip().split(":", 1)
-                queryset = queryset.filter(
-                    project_id=project.id,
-                    **{EventUser.attr_from_keyword(field): identifier},
-                )
+                if field == "ip":
+                    where_conditions.append(
+                        Condition(
+                            Function(
+                                "or",
+                                parameters=[
+                                    Function(
+                                        "equals",
+                                        parameters=[
+                                            Column("ip_address_v4"),
+                                            Function("IPv4StringToNum", parameters=[identifier]),
+                                        ],
+                                    ),
+                                    Function(
+                                        "equals",
+                                        parameters=[
+                                            Column("ip_address_v6"),
+                                            Function("IPv6StringToNum", parameters=[identifier]),
+                                        ],
+                                    ),
+                                ],
+                            ),
+                            Op.EQ,
+                            1,
+                        )
+                    )
+                else:
+                    where_conditions.append(
+                        Condition(
+                            Column(QUERY_TO_SNUBA_FIELD_MAPPING.get(field, field)),
+                            Op.EQ,
+                            identifier,
+                        )
+                    )
             except (ValueError, KeyError):
                 return Response([])
-
+        query = Query(
```

Comment:
I think ideally we could figure out how to share this query with https://github.com/getsentry/sentry/blob/cc7731f92bce347b0bf932c2ffe0b96cca9eaecd/src/sentry/utils/eventuser.py#L86-L92

We also need to figure out a way to handle the unique de-duping here: https://github.com/getsentry/sentry/blob/cc7731f92bce347b0bf932c2ffe0b96cca9eaecd/src/sentry/utils/eventuser.py#L179

Potentially we could just overfetch and de-dupe that way? It'd make the paginator more complicated though

Reply:
yeah there's quite a bit of cleanup that needs to be done in this pr. 

i'm also a bit unclear about what you're suggesting; do we want to use `for_projects` for the endpoint?

Reply:
I'm not totally sure what the solution is, but Nisanthan has a lot of logic in that pr that we'll need in this endpoint. The query will be the same, and we also need the de-duping logic.

The problem is that we can't write a query in Snuba to get all the unique `EventUser` rows, because ideally we want to group by `user` and then get the most recent email, ip, username etc. We can't use window functions in snuba at the moment, so we have to group by `id, email, ip, username` and then de-dupe in the application code.